### DISCUSSION TITLE
Toggle utility rate plan via API ?

### DISCUSSION THREAD

**Discussion by rob0101** [netzero-labs/netzero#25] [Ideas]
I mostly manage my battery outside of TBC and have injection of utility (Amber) prices turned off (no point downloading every 5m if not being used).

Amber sells into price spikes automatically, but is forced by Tesla to stop at 20% SOC.

Occasionally I need to turn Netzero price injection on so that the battery can sell down nearer to 0%.

It'd be nice if I could use the Netzero API to re-enable price fetching, so actions could be automated.

**Comment by zigam** [comment]
Switching between TBC and self-powered will effectively toggle the utility rate plan. Data will still be downloaded (no harm there), but in self-powered mode the rate plan has no effect.

**Reply by rob0101** [reply]
Thanks.

I'll keep toggling it manually for now. I'm not a fan of polling services unnecessarily.
